packages/passport/sdk/src/zkEvm/types.ts

```diff
@@ -111,3 +111,68 @@ export type AccountsChangedEvent = Array<string>;
 export interface ProviderEventMap extends Record<string, any> {
   [ProviderEvent.ACCOUNTS_CHANGED]: [AccountsChangedEvent];
 }
+
+export interface EIP1193Provider {
+  request(args: EIP1193RequestArguments): Promise<unknown>;
+  on<TEvent extends keyof EIP1193EventMap>(event: TEvent, listener: EIP1193EventMap[TEvent]): void;
+  removeListener<TEvent extends keyof EIP1193EventMap>(event: TEvent, listener: EIP1193EventMap[TEvent]): void;
+}
+
+interface EIP1193RequestArguments {
+  readonly method: string;
+  readonly params?: readonly unknown[] | object;
+}
+
+export class ProviderRpcError extends Error {
+  code: number;
+
+  details: string;
+
+  constructor(code: number, message: string) {
+    super(message);
+    this.code = code;
+    this.details = message;
+  }
+}
+
+export type ProviderConnectInfo = {
+  chainId: string
+};
+
+export type ProviderMessage = {
+  type: string
+  data: unknown
+};
+
+export type EIP1193EventMap = {
+  accountsChanged(accounts: string[]): void
+  chainChanged(chainId: string): void
+  connect(connectInfo: ProviderConnectInfo): void
+  disconnect(error: ProviderRpcError): void
+  message(message: ProviderMessage): void
+};
```

Comment:
These types don't seem to be used anywhere

Reply:
These types are used inside the `announceProvider` function found in `zkEVM/provider/eip6963.ts`

Reply:
These are very similar to other type definitions that we have within the same file. Is it not possible to update `EIP6963ProviderDetail` to use the existing `Provider` type? That way, we only need to add interfaces for `EIP6963ProviderDetail`, `EIP6963ProviderInfo` & `EIP6963AnnounceProviderEvent`.